modules/core/CMakeLists.txt

```diff
@@ -168,6 +168,10 @@ if(HAVE_HPX)
   ocv_target_link_libraries(${the_module} LINK_PRIVATE "${HPX_LIBRARIES}")
 endif()
 
+if(HAVE_OPENMP AND DEFINED OpenMP_CXX_INCLUDE_DIRS AND OpenMP_CXX_LIBRARIES)
```

Comment:
> OpenMP_CXX_INCLUDE_DIRS AND OpenMP_CXX_LIBRARIES

mess of different variables

Reply:
My idea is linking `${OpenMP_CXX_LIBRARIES}` when `OpenMP_CXX_INCLUDE_DIRS` is defined (which implies clang environment) and `OpenMP_CXX_LIBRARIES` is not null. Do you suggest using `OpenMP_CXX_LIBRARIES` in replace of `OpenMP_CXX_INCLUDE_DIRS` for definition checking?

Reply:
No need to use other variables here.
Logic should protect usage from empty/non-defined values. No more, no less.

```
if(HAVE_OPENMP AND DEFINED OpenMP_CXX_LIBRARIES AND OpenMP_CXX_LIBRARIES)
  ocv_target_link_libraries(${the_module} LINK_PRIVATE "${OpenMP_CXX_LIBRARIES}")
endif()
```

Reply:
Okay, done with latest commit.

Reply:
Okay, done with latest commit.